internal/database/cart_repository.go

```diff
@@ -0,0 +1,194 @@
+	"innogocartapi/internal/models"
+	"log/slog"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+)
+
+type cartRepository struct {
+	db *sqlx.DB
+}
+
+func NewPostgreConnection(cartConfig config.CartConfiguration) (*sqlx.DB, error) {
+	db, err := sqlx.Open("postgres", fmt.Sprintf("host=%s port=%s dbname=%s user=%s password=%s sslmode=%s",
+		cartConfig.DbHost,
+		cartConfig.DbPort,
+		cartConfig.Name,
+		cartConfig.User,
+		cartConfig.Pass,
+		cartConfig.SSLMode))
+	if err != nil {
+		return db, err
+	}
+
+	return db, nil
+}
+
+func NewPostgreRepository(db *sqlx.DB) *cartRepository {
+	return &cartRepository{
+		db: db,
+	}
+}
+
+func (repo *cartRepository) CreateCart(ctx context.Context) (models.Cart, error) {
+	var resultCartDb Cart
+
+	rows, err := repo.db.QueryContext(ctx, "insert into carts (id) Values(default) returning id")
+
+	if err != nil {
+		slog.Error("failed to create cart", "error", err)
+		return models.Cart{}, domainErrs.ErrInternal
+	}
+
+	for rows.Next() {
+		err = rows.Scan(&resultCartDb.Id)
+		if err != nil {
+			slog.Error("failed to scan created cart", "error", err)
+			return models.Cart{}, domainErrs.ErrInternal
+		}
+	}
+
+	return cartDbToCart(resultCartDb), nil
+}
+
+func (repo *cartRepository) GetCart(ctx context.Context, cartId models.CartId) (models.Cart, error) {
+	var resultCartDb Cart
+	var resultCartItemsDb []CartItem
+
+	err := repo.db.GetContext(
+		ctx,
+		&resultCartDb,
+		`select id from carts 
+		    where id = $1
+		limit 1`,
+		cartId)
+	if err != nil {
+		switch err {
+		case sql.ErrNoRows:
+			return models.Cart{}, domainErrs.ErrNotFound
+		default:
+			slog.Error("failed to get cart", "error", err)
+			return models.Cart{}, domainErrs.ErrInternal
+		}
+	}
+
+	err = repo.db.SelectContext(
+		ctx,
+		&resultCartItemsDb,
+		`select cart_id, id, product, quantity from cart_items 
+			where cart_id = $1`,
+		resultCartDb.Id)
+
+	if err != nil {
+		slog.Error("failed to get cart items", "error", err)
+		return models.Cart{}, domainErrs.ErrInternal
+	}
+
+	resultCart := cartDbToCart(resultCartDb)
+	resultCart.Items = make([]models.CartItem, 0, len(resultCartItemsDb))
+
+	for _, cartItemDb := range resultCartItemsDb {
+		resultCart.Items = append(resultCart.Items, cartItemDbToCartItem(cartItemDb))
+	}
+
+	return resultCart, nil
+}
+
+func (repo *cartRepository) RemoveFromCart(ctx context.Context, cartItemId models.CartItemId) error {
+	_, err := repo.db.ExecContext(
+		ctx,
+		`delete from cart_items 
+			where id = $1`,
+		cartItemId)
+	if err != nil {
+		slog.Error("failed to remove from cart", "error", err)
+		return domainErrs.ErrInternal
+	}
+
+	return nil
+}
+
+func (repo *cartRepository) AddToCart(ctx context.Context, cartItem models.CartItem) (models.CartItem, error) {
+	var resultCartItemDb CartItem
+
+	rows, err := repo.db.QueryContext(
+		ctx,
+		`insert into cart_items(cart_id, product, quantity) 
+			values($1, $2, $3) returning cart_id, id, product, quantity`,
```

Comment:
I think you can stay with this:

```suggestion
			values($1, $2, $3) returning id`,
```

There is no need to return the rest